### PR TITLE
Fix the tests since webexport file hosting is no longer available.

### PR DESCRIPTION
**Pull request by Hammarn**
The commit history is a bit messy since I accidentally added this to a branch where I had merged in PR #196.
I reverted them and could rebase them if we want a prettier history, but this is more honest. 

The fasta file is used to generate the star-index instead of including it in the testdata. This makes the test take a bit longer but it saves enough space to include the test data in the repo itself instead of relying on external file hosting.

### CODE REVIEW

**Comment by ewels** [comment]
+1 for rebasing 😆

**Comment by Hammarn** [comment]
Rebased, a little less hideous now @ewels

File: tests/amazon_test.sh [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+if [ -z $1] 
 then
     echo "No argument given, going to try to run ../main.nf"
 else
     script_path=$1
 fi
 
-data_path="/tmp"
+data_path="./test_data"
 if [ -d "./test_data" ]
```
Comment (ewels) [inline tests/amazon_test.sh]: You can remove this `if` statement - all it does is set `data_path` to `./test_data` if `./test_data` already exists. `./test_data` is now the default so it has no effect.

File: tests/amazon_test.sh [hunk 2]
```diff
@@ -1,14 +1,14 @@
 #!/usr/bin/env bash
 
 script_path="../main.nf"
-if [ -z $1]
+if [ -z $1]
```
Comment (ewels) [inline tests/amazon_test.sh]: 🤓

File: tests/amazon_test.sh [hunk 3]
```diff
@@ -24,14 +24,14 @@ data_dir=${data_path}/ngi-rna_test_set
 if [ -d $data_dir ]
 then
     echo "Found existing test set, using $data_dir"
-else
-    echo "Downloading test set..."
-    curl https://export.uppmax.uu.se/b2013064/test-data/ngi-rna_test_set.tar.bz2 > ${data_path}/ngi-rna_test_set.tar.bz2
+else
```
Comment (ewels) [inline tests/amazon_test.sh]: whitespace 🤓

File: tests/amazon_test.sh [hunk 4]
```diff
@@ -24,14 +24,14 @@ data_dir=${data_path}/ngi-rna_test_set
...
-    curl https://export.uppmax.uu.se/b2013064/test-data/ngi-rna_test_set.tar.bz2 > ${data_path}/ngi-rna_test_set.tar.bz2
+else    
     echo "Unpacking test set..."
-    tar xvjf ${data_path}/ngi-rna_test_set.tar.bz2 -C ${data_path}
+    mkdir -p ${data_path}
+    tar xvjf ngi-rna_test_set.tar.bz2 -C ${data_path}
     echo "Done"
 fi
 
+run_name="Test RNA Run: "$(date +%s)
```
Comment (ewels) [inline tests/amazon_test.sh]: I don't think that this is used in this script?

File: tests/docker_test.sh [hunk 5]
```diff
@@ -8,7 +8,7 @@ else
     script_path=$1
 fi
 
-data_path="/tmp"
+data_path="./test_data"
 if [ -d "./test_data" ]
```
Comment (ewels) [inline tests/docker_test.sh]: As above

File: tests/uppmax_test.sh [hunk 6]
```diff
@@ -1,14 +1,14 @@
 #!/usr/bin/env bash
 
 script_path="../main.nf"
-if [ -z $1]
+if [ -z $1]
```
Comment (ewels) [inline tests/uppmax_test.sh]: 🤓

File: tests/uppmax_test.sh [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
+if [ -z $1] 
 then
     echo "No argument given, going to try to run ../main.nf"
 else
     script_path=$1
 fi
 
-data_path=$SNIC_NOBACKUP
+data_path="./test_data"
 if [ -d "./test_data" ]
```
Comment (ewels) [inline tests/uppmax_test.sh]: As above

File: tests/uppmax_test.sh [hunk 8]
```diff
@@ -17,23 +17,22 @@ fi
 
 curl --version >/dev/null 2>&1 || { echo >&2 "I require curl, but it's not installed. Aborting."; exit 1; }
 tar --version >/dev/null 2>&1 || { echo >&2 "I require tar, but it's not installed. Aborting."; exit 1; }
+docker -v >/dev/null 2>&1 || { echo >&2 "I require docker, but it's not installed. Visit https://www.docker.com/products/overview#/install_the_platform  ."; exit 1; }
```
Comment (ewels) [inline tests/uppmax_test.sh]: Docker for Uppmax?

File: tests/uppmax_test.sh [hunk 9]
```diff
@@ -17,23 +17,22 @@ fi
...
 nextflow -v >/dev/null 2>&1 || { echo >&2 "I require nextflow, but it's not installed. If you hava Java, run 'curl -fsSL get.nextflow.io | bash'. If not, install Java."; exit 1; }
 
 data_dir=${data_path}/ngi-rna_test_set
 if [ -d $data_dir ]
 then
     echo "Found existing test set, using $data_dir"
-else
-    echo "Downloading test set..."
-    curl https://export.uppmax.uu.se/b2013064/test-data/ngi-rna_test_set.tar.bz2 > ${data_path}/ngi-rna_test_set.tar.bz2
+else
```
Comment (ewels) [inline tests/uppmax_test.sh]: 🤓

File: tests/uppmax_test.sh [hunk 10]
```diff
@@ -17,23 +17,22 @@ fi
...
-    tar xvjf ${data_path}/ngi-rna_test_set.tar.bz2 -C ${data_path}
+    mkdir -p ${data_path}
+    tar xvjf ngi-rna_test_set.tar.bz2 -C ${data_path}
     echo "Done"
 fi
 
 run_name="Test RNA Run: "$(date +%s)
-
-cmd="nextflow run $script_path -resume -name \"$run_name\" -profile uppmax_devel --gtf ${data_dir}/genes.gtf --bed12 ${data_dir}/genes.bed --star_index ${data_dir}/star/ --singleEnd --reads \"${data_dir}/*.fastq.gz\""
+cmd="nextflow run $script_path -resume -name \"$run_name\" -profile uppmax_devel --gtf ${data_dir}/genes.gtf --bed12 ${data_dir}/genes.bed --fasta ${data_dir}/genome.fa/ --singleEnd --reads \"${data_dir}/*.fastq.gz\""
```
Comment (ewels) [inline tests/uppmax_test.sh]: You have a trailing slash on `genome.fa/`

File: tests/uppmax_test_hisat.sh [hunk 11]
```diff
@@ -1,14 +1,14 @@
 #!/usr/bin/env bash
 
 script_path="../main.nf"
-if [ -z $1]
+if [ -z $1]
```
Comment (ewels) [inline tests/uppmax_test_hisat.sh]: 🤓

File: tests/uppmax_test_hisat.sh [hunk 12]
```diff
@@ -1,14 +1,14 @@
...
+if [ -z $1] 
 then
     echo "No argument given, going to try to run ../main.nf"
 else
     script_path=$1
 fi
 
-data_path=$SNIC_NOBACKUP
+data_path="./test_data"
 if [ -d "./test_data" ]
```
Comment (ewels) [inline tests/uppmax_test_hisat.sh]: As above

File: tests/uppmax_test_hisat.sh [hunk 13]
```diff
@@ -17,22 +17,21 @@ fi
 
 curl --version >/dev/null 2>&1 || { echo >&2 "I require curl, but it's not installed. Aborting."; exit 1; }
 tar --version >/dev/null 2>&1 || { echo >&2 "I require tar, but it's not installed. Aborting."; exit 1; }
+docker -v >/dev/null 2>&1 || { echo >&2 "I require docker, but it's not installed. Visit https://www.docker.com/products/overview#/install_the_platform  ."; exit 1; }
```
Comment (ewels) [inline tests/uppmax_test_hisat.sh]: Again, docker won't be installed on Uppmax

File: tests/uppmax_test_hisat.sh [hunk 14]
```diff
@@ -17,22 +17,21 @@ fi
...
 nextflow -v >/dev/null 2>&1 || { echo >&2 "I require nextflow, but it's not installed. If you hava Java, run 'curl -fsSL get.nextflow.io | bash'. If not, install Java."; exit 1; }
 
 data_dir=${data_path}/ngi-rna_test_set
 if [ -d $data_dir ]
 then
     echo "Found existing test set, using $data_dir"
-else
-    echo "Downloading test set..."
-    curl https://export.uppmax.uu.se/b2013064/test-data/ngi-rna_test_set.tar.bz2 > ${data_path}/ngi-rna_test_set.tar.bz2
+else
```
Comment (ewels) [inline tests/uppmax_test_hisat.sh]: 🤓

File: tests/uppmax_test_hisat.sh [hunk 15]
```diff
@@ -17,22 +17,21 @@ fi
...
     echo "Unpacking test set..."
-    tar xvjf ${data_path}/ngi-rna_test_set.tar.bz2 -C ${data_path}
+    mkdir -p ${data_path}
+    tar xvjf ngi-rna_test_set.tar.bz2 -C ${data_path}
     echo "Done"
 fi
 
 run_name="Test RNA Run: "$(date +%s)
-
 cmd="nextflow run $script_path -resume -name \"$run_name\" -profile uppmax_devel --gtf ${data_dir}/genes.gtf --bed12 ${data_dir}/genes.bed --hisat2_index ${data_dir}/r64/ --aligner hisat2 --singleEnd --reads \"${data_dir}/*.fastq.gz\""
```
Comment (ewels) [inline tests/uppmax_test_hisat.sh]: Are we packaging the hisat2 index too? We can remove this as well.

Comment (Hammarn) [reply tests/uppmax_test_hisat.sh]: We could, but the hisat2 index is so tiny comparatively so it doesn't really make a difference in my opinion.

**Comment by ewels** [comment]
Tests passing! 🎉

**Pull request merged**